### ISSUE TITLE
Missing test coverage for SSL client endpoint string description parser

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5801]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5801|
|Type|defect|
|Created|2012-07-24 12:09:09Z|
|Branch|https://github.com/twisted/twisted/tree/ssl-client-endpoint-parser-test-5801-2|

The default for `certKey` is untested, as is the default for `sslmethod`.

Attachments:

* [ssl-client-endpoint-test-defaults-5801.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/fd3/fd39816994d79b4b2cbc07d6eaaa0864c01239e8/204f086092747a97c31c6748f921b17321cb1ee0.patch) (1620 bytes) - added by ecnahc515 on 2013-03-09 23:43:16Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5801 5801
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_ssl_client_endpoint_parser_test_5801_2 branches/ssl-client-endpoint-parser-test-5801-2
branch_author__ashfall__tomprince ashfall, tomprince
status__closed closed
resolution__fixed fixed
component__core core
keywords__endpoints_easy endpoints easy
time__1343131749000000 1343131749000000
changetime__1364322467000000 1364322467000000
version__None None
owner__tom_prince tom.prince

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_1"></a>|@ashfall set owner to @ashfall|
|-|-|

This looks like something I do a lot.

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_2"></a>|@ashfall commented|
|-|-|

(In [34995]) Branching to 'ssl-client-endpoint-parser-test-5801'

**Comment by twisted-trac** [comment]
|<img alt="ecnahc515's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|ecnahc515 removed owner|
|-|-|

Added a new test which tests that default values for the ssl client endpoint string parser are set.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince commented|
|-|-|

(In [37818]) Branching to ssl-client-endpoint-parser-test-5801-2.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince set owner to @tomprince|
|-|-|

[build results](https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/ssl-client-endpoint-parser-test-5801-2)

This looks good. I've slightly expanded the docstring, to be more precise. I'll merge this.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_6"></a>|@tomprince commented|
|-|-|

And, it needs a [topfile](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_7"></a>|@tomprince set status to `closed`|
|-|-|

(In [37824]) Merge ssl-client-endpoint-parser-test-5801-2: Add missing test coverage for SSL client endpoint string description parser

Author: ecnahc515
Reviewers: tom.prince
Fixes: [#5801](https://github.com/twisted/twisted/issues/5801)
